nmostesting/mocks/Auth.py

```diff
@@ -0,0 +1,535 @@
+
+        # hmm, no client authorization done, just redirects a random authorization code back to the client
+        # TODO: add web pages for client authorization for the future
+
+        # "If the request fails due to a missing, invalid, or mismatching
+        # redirection URI, or if the client identifier is missing or invalid,
+        # the authorization server SHOULD inform the resource owner of the
+        # error and MUST NOT automatically redirect the user-agent to the
+        # invalid redirection URI."
+        # see https://tools.ietf.org/html/rfc6749#section-4.1.2.1
+        error = None
+        error_description = None
+        redirect_uri = REDIRECT_URI
+        if "redirect_uri" in request.args:
+            redirect_uri = request.args["redirect_uri"]
+            # TODO: check is the redirect_uri in the registered redirect_uris
+        if not redirect_uri:
+            raise AuthException("invalid_request", "missing redirect_uri")
+
+        if "client_id" not in request.args:
+            raise AuthException("invalid_request", "missing client_id")
+
+        if "response_type" not in request.args:
+            error = "invalid_request"
+            error_description = "missing response_type"
+        else:
+            response_type = request.args["response_type"]
+            if response_type != RESPONSE_TYPES.code.name:
+                error = "invalid_request"
+                error_description = "response_type not code"
+
+        if "scope" in request.args:
+            valid_scope_found = False
+            scopes = request.args["scope"].split()
+            for scope in scopes:
+                if scope in SCOPES.__members__:
+                    valid_scope_found = True
+                    break
+            if not valid_scope_found:
+                error = "invalid_request"
+                error_description = "scope: {} are not supported".format(scopes)
+
+        # create a random authorization code
+        # TODO: use it to test when client exchanges it for a bearer token
+        auth.code = str(uuid.uuid4())
+        vars = {"code": auth.code}
+
+        if "state" in request.args:
+            vars["state"] = request.args["state"]
+
+        if error:
+            vars["error"] = error
+
+        if error_description:
+            vars["error_description"] = error_description
+
+        return redirect("{}?{}".format(redirect_uri, urllib.parse.urlencode(vars)))
+
+    except AuthException as e:
+        error_message = {"code": e.httpstatus.value,
+                         "error": e.error,
+                         "error_description": e.description}
+        return Response(json.dumps(error_message), status=e.httpstatus.value, mimetype='application/json')
+
+
+@ AUTH_API.route('/token', methods=["POST"])
+def auth_token():
+    auth = AUTHS[flask.current_app.config["AUTH_INSTANCE"]]
+    try:
+        auth_header_required = False
+        scopes = []
+        request_data = request.get_data()
```

Comment:
Should we check/require `Content-Type` is `application/x-www-form-urlencoded`?

Reply:
Check added to the `/token`, and the `/authorize` endpoints